compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingOps.cpp

```diff
@@ -70,8 +70,25 @@ LogicalResult UnsetEncodingOp::verify() {
   if (getResultType().getRank() != getSourceType().getRank()) {
     return emitOpError("cannot change the rank of the tensor");
   }
-  if (failed(verifyCompatibleShape(getResultType(), getSourceType()))) {
-    return emitOpError("expected to preserve the logical shape of the tensor");
+
+  ArrayRef<int64_t> srcShape = getSourceType().getShape();
+  ArrayRef<int64_t> resShape = getResultType().getShape();
+  for (auto [idx, src, res] : llvm::enumerate(srcShape, resShape)) {
+    if (!ShapedType::isDynamic(src) && !ShapedType::isDynamic(res) &&
+        src < res) {
+      return emitOpError()
+             << "expected to preserve the logical shape of the tensor: the "
+             << idx << "-th dimension has source shape " << src
```

Comment:
wording: "shape" suggests an array, but you're printing a single int64_t value `src`. Maybe call that a "size" ?

Reply:
resolving, this part of the diff is in the base PR 19023, moved comment there.